packages/next/next-server/server/image-optimizer.ts

```diff
@@ -43,27 +44,31 @@ export async function imageOptimizer(
     return { finished: true }
   }
 
-  let absoluteUrl: URL | undefined
-  let relativeUrl: string | undefined
+  let isAbsolute: boolean
 
   if (url.startsWith('/')) {
-    relativeUrl = url
+    href = url
+    isAbsolute = false
   } else {
+    let hrefParsed: URL
+
     try {
-      absoluteUrl = new URL(url)
+      hrefParsed = new URL(url)
+      isAbsolute = true
+      href = hrefParsed.toString()
```

Comment:
I'm not sure if toString() could ever throw an error, but if so it might be better to put `isAbsolute = true` below it

Reply:
We return below this if it errors so `isAbsolute` won't be used after this if it does throw